### PR TITLE
pins: improve error reporting for sss pin

### PR DESCRIPTION
**Pull request by sergio-correia**
clevis-encrypt-sss would return 0 (and a JWE, without the pins that failed)
even if the encryption operation failed, when using the sss pin, such as:

echo test | clevis encrypt sss '{"t":1, "pins":{"tang":[{"url":"foo bar"}]}}'
Unable to fetch advertisement: 'foo bar/adv/'!
eyJhbGciOiJkaXIiLCJjbGV2aXMiOnsicGluIjoic3NzIiwic3NzIjp7Imp3ZSI6WyIiXSwicCI6Inlqald0bkM2ZkwtNDhIbTd5Yy1fVmlwWWxTTHZ1ZlliZnRaejFjLW5ucU0iLCJ0IjoxfX0sImVuYyI6IkEyNTZHQ00ifQ..rM6IMXjNIN3uybW1.J5jAtiU.61x_fjXJC1uCjljTnkkdCw

echo $?
0

This is particularly problematic when using clevis luks bind, as it would
appear the operation succeeded, while in practice it did not, which
means we would be unable to unlock this device. E.g.:

clevis luks bind -d device sss '{"t":1, "pins":{"tang":[{"url":"foo bar"}]}}' <<< password
Unable to fetch advertisement: 'foo bar/adv/'!

echo $?
0

clevis luks list -d device
1: sss '{"t":1,"pins":{}}'

clevis luks unlock -d device
Nothing to read on input.

We now check the exit code of the pins encryption, when doing an sss
encrypt, and indicate errors correctly, so that we avoid this situation.

After the fix:

echo test | clevis encrypt sss '{"t":1, "pins":{"tang":[{"url":"foo bar"}]}}'
Unable to fetch advertisement: 'foo bar/adv/'!

echo $?
1

Also:
clevis luks bind -d device sss '{"t":1, "pins":{"tang":[{"url":"foo bar"}]}}' <<< password
Unable to fetch advertisement: 'foo bar/adv/'!

echo $?
1

clevis luks list -d device
[no output]

### CODE REVIEW

File: src/pins/sss/meson.build [hunk 1]
```diff
@@ -18,8 +18,9 @@ if jansson.found() and libcrypto.found()
   mans += join_paths(src, 'clevis-encrypt-sss.1')
 
   env = environment()
-  env.append('PATH',
+  env.prepend('PATH',
```
Comment (jlebon) [inline src/pins/sss/meson.build:21]: Is this change related to the rest of the patch? It looks like it's so that we use the in-tree scripts instead of the already installed ones? Might be cleaner as a separate commit with its own commit message.

File: src/luks/tests/bad-sss [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+TEST=$(basename "${0}")
+. tests-common-functions
+
+on_exit() {
+    [ -d "${TMP}" ] && rm -rf "${TMP}"
+}
+
+trap 'on_exit' EXIT
+trap 'exit' ERR
```
Comment (jlebon) [inline src/luks/tests/bad-sss:29]: Is this line actually needed? Errors should cause exits already in `set -e` mode.

Comment (sergio-correia) [reply src/luks/tests/bad-sss:29]: Good catch. It's not required, indeed. Let me remove this.

File: src/pins/sss/meson.build (hunk 1 above)
Comment (sergio-correia) [reply src/pins/sss/meson.build:21]: You are right. Let me split this part.
